backend/pkg/api/instances.go

```diff
@@ -631,3 +639,133 @@ func (api *API) instanceStatusHistoryQuery(instanceID, appID, groupID string, li
 		Order(goqu.C("created_ts").Desc()).
 		Limit(uint(limit))
 }
+
+// instanceFactQuery returns a SelectDataset prepared to return all instances
+// that have been checked in during a given duration from a given time.
+func (api *API) instanceFactQuery(t *time.Time, duration *time.Duration) *goqu.SelectDataset {
+	if t == nil {
+		now := time.Now()
+		t = &now
+	}
+
+	if duration == nil {
+		d := time.Hour
+		duration = &d
+	}
+
+	// Convert duration to string
+	interval := fmt.Sprintf("%d microseconds", int(duration.Microseconds()))
+
+	timestamp := goqu.L("timestamp ?", t.Format(time.RFC3339))
+	timestampMinusDuration := goqu.L("timestamp ? - interval ?", t.Format(time.RFC3339), interval)
+
+	query := goqu.From(goqu.T("instance_application")).
+		Select(
+			goqu.C("last_check_for_updates").As("timestamp"),
+			goqu.L("channel.name").As("channel_name"),
+			goqu.Case().
+				When(goqu.L("channel.arch").Eq(1), "AMD64").
+				When(goqu.L("channel.arch").Eq(2), "ARM").
+				As("arch"),
+			goqu.L("package.version").As("version"),
+			goqu.L("COUNT(*)").As("instances")).
+		Join(goqu.L("groups"), goqu.On(goqu.L("group_id").Eq(goqu.L("groups.id")))).
+		Join(goqu.L("channel"), goqu.On(goqu.L("groups.channel_id").Eq(goqu.L("channel.id")))).
+		Join(goqu.L("package"), goqu.On(goqu.L("channel.package_id").Eq(goqu.L("package.id")))).
+		Where(
+			goqu.L("last_check_for_updates").Gt(timestampMinusDuration),
+			goqu.L("last_check_for_updates").Lte(timestamp)).
+		GroupBy(goqu.L("last_check_for_updates"),
+			goqu.L("channel.name"),
+			goqu.L("channel.arch"),
+			goqu.L("package.version")).
+		Order(goqu.L("last_check_for_updates").Asc())
+
+	// // Print input parameters
+	// fmt.Printf("Input time: %s\n", t.Format(time.RFC3339))
+	// fmt.Printf("Input duration: %s\n", interval)
+
+	// // Print the generated SQL
+	// sql, _, _ := query.ToSQL()
+	// fmt.Printf("Generated SQL: %s\n", sql)
+
+	return query
+}
+
+// GetInstanceFacts returns an InstanceFact table with all instances that have
+// been previously been checked in.
+func (api *API) GetInstanceFacts() ([]InstanceFact, error) {
+	query := goqu.From("instance_fact").
+		Select(goqu.L("*")).Order(goqu.C("timestamp").Asc()).
```

Comment:
```suggestion
		Select(goqu.L("*")).
```
Already done in the line below